habitat/tasks/rearrange/rearrange_grasp_manager.py

```diff
@@ -125,23 +239,21 @@ def snap_to_obj(self, snap_obj_id: int, force: bool = True):
             CollisionGroups.UserGroup7
         )
 
-        def create_hold_constraint(pivot_in_link, pivot_in_obj):
-            c = RigidConstraintSettings()
-            c.object_id_a = self._sim.robot.get_robot_sim_id()
-            c.link_id_a = self._sim.robot.ee_link_id
-            c.object_id_b = self._snapped_obj_id
-            c.pivot_a = pivot_in_link
-            c.pivot_b = pivot_in_obj
-            c.max_impulse = self._config.GRASP_IMPULSE
-            return self._sim.create_rigid_constraint(c)
-
         self._snap_constraints = [
-            create_hold_constraint(mn.Vector3(0.1, 0, 0), mn.Vector3(0, 0, 0)),
-            create_hold_constraint(
-                mn.Vector3(0.0, 0, 0), mn.Vector3(-0.1, 0, 0)
+            self.create_hold_constraint(
+                mn.Vector3(0.1, 0, 0),
+                mn.Vector3(0, 0, 0),
+                self._snapped_obj_id,
+            ),
+            self.create_hold_constraint(
+                mn.Vector3(0.0, 0, 0),
+                mn.Vector3(-0.1, 0, 0),
+                self._snapped_obj_id,
             ),
-            create_hold_constraint(
-                mn.Vector3(0.1, 0.0, 0.1), mn.Vector3(0.0, 0.0, 0.1)
+            self.create_hold_constraint(
+                mn.Vector3(0.1, 0.0, 0.1),
+                mn.Vector3(0.0, 0.0, 0.1),
+                self._snapped_obj_id,
```

Comment:
The fixed constraint type is working nicely these days for 6dof constraints, so maybe we can git rid of this hack and just use that.

Reply:
Yes, let's address this in future PR.